src/guide/components.md

```diff
@@ -135,9 +135,25 @@ var MyComponent = Vue.extend({
 
 The `el` option also requires a function value when used in `Vue.extend()`, for exactly the same reason.
 
-### `is` attribute
+### Template Caveats
 
-Some HTML elements, for example `<table>`, have restrictions on what elements can appear inside them. Custom elements that are not in the whitelist will be hoisted out and thus not render properly. In such cases you should use the `is` special attribute to indicate a custom element:
+Because Vue.js templates are not string-based, but rather DOM-based, they have to be individually valid pieces of HTML. Some HTML elements have restrictions on what elements can appear inside them. Most common of these restrictions are:
+
+- `a` can not contain other interactive elements (e.g. buttons and other links)
+- `li` should be a direct child of `ul`, and `ul` can only contain `li`
+- `option` should be a direct child of `select`, and `select` can only contain `option` (and `optgroup`)
+- `table` can only contain `thead`, `tbody`, `tfoot` and `tr`, and these elements should be direct cildren of `table`
+- `tr` can only contain `th` and `td`, and these elements should be direct children of `tr`
+
+In practive these restriction can cause unexpected behavior. Although in simple cases it might appear to work, you can not rely on custom elements being expanded before browser validation. E.g. `<my-select><option>...</option></my-select>` is not a valid template even if `my-select` component eventually expands to `<select>...</select>`.
+
+Another consequence is that you can not use custom elements inside of `ul`, `select`, `table` and other elements with similar  restrictions. Custom elements will be hoisted out and thus not render properly. In such cases you should use the `is` special attribute to indicate a custom element:
+
+``` html
```

Comment:
I'd add several examples of incorrect html with appropriate correct examples (suggestions how to make it correct)